src/stream-tcp-reassemble.c

```diff
@@ -1846,7 +1846,7 @@ int StreamTcpReassembleHandleSegment(ThreadVars *tv, TcpReassemblyThreadCtx *ra_
     } else if (p->tcph->th_flags & TH_RST) { // accepted rst
         dir = UPDATE_DIR_PACKET;
     } else if ((p->tcph->th_flags & TH_FIN) && ssn->state > TCP_TIME_WAIT) {
-        dir = UPDATE_DIR_PACKET;
+        dir = UPDATE_DIR_BOTH;
```

Comment:
In the commit you mention this is for FIN+ACK, so should this only run if TH_ACK flag is also set?

Reply:
Uploaded new PR: https://github.com/OISF/suricata/pull/6686